### ISSUE TITLE
Excessive logging in Redis server during actor reconstruction.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#807]
Consider the following script.

```python
import ray
import time

ray.worker._init(start_ray_local=True, num_local_schedulers=2, num_workers=0, redirect_output=True)

@ray.remote
class Foo(object):
    def __init__(self):
        pass
    def plasma(self):
        return ray.worker.global_worker.plasma_client.store_socket_name


local_plasma = ray.worker.global_worker.plasma_client.store_socket_name

# Create an actor on a different local scheduler.
a = Foo.remote()
while ray.get(a.plasma.remote()) == local_plasma:
    a = Foo.remote()

# Kill the local scheduler that the actor is on.
ray.services.all_processes[ray.services.PROCESS_TYPE_LOCAL_SCHEDULER][1].kill()

time.sleep(1)

# Schedule a task on the actor and get the result.
ray.get(a.plasma.remote())
```

The script runs fine (it takes ten seconds for the monitor to detect that the local scheduler has died). But one of the Redis shards will continuously log the following message (you can see this more vividly if you omit the `redirect_output=True`).

```
WARN] (/Users/rkn/Workspace/ray/src/common/redis_module/ray_redis_module.cc:977) No subscribers received this publish. This most likely means that either the intended recipient has not subscribed yet or that the pubsub connection to the intended recipient has been broken.
```

The problem is that the initial publish (when the task is submitted) is not received by the dead local scheduler, and so the same publish is reattempted **infinitely**.

The retrying is done [here](https://github.com/ray-project/ray/blob/37282330c0ea687fd1b983176dce85731fcf189d/src/common/state/redis.cc#L884) and [here](https://github.com/ray-project/ray/blob/37282330c0ea687fd1b983176dce85731fcf189d/src/common/state/redis.cc#L952).

The logging is done [here](https://github.com/ray-project/ray/blob/37282330c0ea687fd1b983176dce85731fcf189d/src/common/redis_module/ray_redis_module.cc#L974).

**Comment by robertnishihara** [comment]
I think this issue should be partially addressed by #894. At least the logging should no longer be infinite.

**Comment by robertnishihara** [comment]
This has been fixed, probably by #1668.
